src/preset.js

```diff
@@ -8,25 +8,153 @@
+ */
+genotet.preset.NETWORK_NAME_ = 'My Network';
+/**
+ * Name for preset expression view.
+ * @private @const {string}
+ */
+genotet.preset.EXPRESSION_NAME_ = 'My Expression Matrix';
+/**
+ * Name for preset binding view.
+ * @private @const {string}
+ */
+genotet.preset.BINDING_NAME_ = 'My Genome Browser';
+/**
+ * Name for preset 3-track binding view.
+ * @private @const {string}
+ */
+genotet.preset.THREE_TRACK_BINDING_NAME_ = 'My 3 Tracks Genome Browser';
+
+/**
  * Loads a preset with the given name.
  * @param {string} preset
  */
 genotet.preset.loadPreset = function(preset) {
+  genotet.viewManager.closeAllViews();
+  var expressionType = genotet.data.VIEW_TYPES[0];
+  var networkType = genotet.data.VIEW_TYPES[1];
+  var bindingType = genotet.data.VIEW_TYPES[2];
+
   if (!preset) {
     preset = 'default';
   }
-
   switch (preset) {
     case 'default':
+      genotet.preset.createView_(networkType,
+        genotet.preset.NETWORK_NAME_, genotet.preset.NETWORK_PARAMS_);
+      genotet.preset.createView_(expressionType,
+        genotet.preset.EXPRESSION_NAME_, genotet.preset.EXPRESSION_PARAMS_);
+      genotet.preset.createView_(bindingType,
+        genotet.preset.BINDING_NAME_, genotet.preset.BINDING_PARAMS_);
       break;
     case 'network':
+      genotet.preset.createView_(networkType,
+        genotet.preset.NETWORK_NAME_, genotet.preset.NETWORK_PARAMS_);
+      genotet.preset.createView_(bindingType,
+        genotet.preset.BINDING_NAME_, genotet.preset.BINDING_PARAMS_);
       break;
     case 'expression':
+      genotet.preset.createView_(networkType,
+        genotet.preset.NETWORK_NAME_, genotet.preset.NETWORK_PARAMS_);
+      genotet.preset.createView_(expressionType,
+        genotet.preset.EXPRESSION_NAME_, genotet.preset.EXPRESSION_PARAMS_);
       break;
     case 'binding':
+      genotet.preset.createView_(networkType,
+        genotet.preset.NETWORK_NAME_, genotet.preset.NETWORK_PARAMS_);
+      genotet.preset.createView_(expressionType,
+        genotet.preset.EXPRESSION_NAME_, genotet.preset.EXPRESSION_PARAMS_);
+      genotet.preset.createView_(bindingType,
+        genotet.preset.THREE_TRACK_BINDING_NAME_,
+        genotet.preset.THREE_TRACK_BINDING_PARAMS_);
       break;
     default:
       genotet.error('unknown preset:', preset);
       return;
   }
+  genotet.linkManager.link();
+};
+
+/**
+ * Creates a preset view.
+ * @param {string} type Type for preset view.
+ * @param {string} viewName Name for preset view.
+ * @param {!genotet.ExpressionViewParams|!genotet.NetworkViewParams|
+ *   !genotet.BindingViewParams} params
+ * @private
+ */
+genotet.preset.createView_ = function(type, viewName, params) {
+  genotet.viewManager.createView(type, viewName, params);
+
+  var newView = genotet.viewManager.views[viewName];
+  switch (type) {
+    case 'network':
+      genotet.linkManager.register(newView);
+      break;
+    case 'expression':
+      for (var linkViewName in genotet.linkManager.links) {
```

Comment:
Going back to the problem of how you initialize the links. Because the links themselves are tuples, it would be ideal to initialize them with a list of tuples, and you register each of them.

Right now you are doing all-pair view&action&response which implicitly creates a list of all possible tuples. This is unnecessary. Additionally, it can lead to implicit "partial" linking based on the order you create the views. For example, if you create in order networkA, binding, networkB, then binding will have been linked to networkA but not networkB.